pkg/processor/procbuiltin/unwrap.go

```diff
@@ -93,6 +96,78 @@ func (p *unwrapProcessor) Process(_ context.Context, in record.Record) (record.R
 	return out, nil
 }
 
+/*
+Example of an OpenCDC record:
+{}
+*/
+
+// openCDCUnwrapper unwraps an OpenCDC record from the payload, by unmarhsalling rec.Payload.After into type Record.
+type openCDCUnwrapper struct{}
+
+func (o *openCDCUnwrapper) Unwrap(rec record.Record) (record.Record, error) {
+	// record must be structured
+	structPayload, ok := rec.Payload.After.(record.StructuredData)
+	if !ok {
+		return record.Record{}, cerrors.Errorf("record payload data must be structured data")
+	}
+
+	// get payload
+	structPayload, ok = structPayload["payload"].(map[string]any)
+	if !ok {
+		return record.Record{}, cerrors.Errorf("payload doesn't contain a record")
+	}
+	
+	var r record.Record
+	if err := json.Unmarshal(rec.Payload.After.Bytes(), &r); err != nil {
+		return record.Record{}, cerrors.Errorf("error unmarshalling record: %w", err)
+	}
+
+	return record.Record{
+		Key:      o.UnwrapKey(rec.Key),
+		Position: rec.Position,
+		Metadata: nil,
+		Payload: record.Change{
+			Before: nil,
+			After:  structPayload,
+		},
+		Operation: record.OperationSnapshot,
+	}, nil
+}
+
+// UnwrapKey unwraps key as a kafka connect formatted record, returns the key's payload content, or returns the
+// original key if payload doesn't exist.
+func (k *openCDCUnwrapper) UnwrapKey(key record.Data) record.Data {
```

Comment:
FYI - I didn't really implement UnwrapKey, this was mostly copied from the kafka connect unwrapper